manager/manager.go

```diff
@@ -432,7 +432,7 @@ func (m *Manager) Run(parent context.Context) error {
 		grpc.WithTransportCredentials(m.config.SecurityConfig.ClientTLSCreds),
 	}
 
-	cs := raftpicker.NewConnSelector(m.RaftNode, proxyOpts...)
+	cs := raftpicker.NewConnSelector(ctx, m.RaftNode, proxyOpts...)
```

Comment:
I think instead of using context cancellation to stop the goroutine, `raftpicker` should have a `Stop` method that we call before stopping the raft node.